pkg/driver/driver_openstack.go

```diff
@@ -468,3 +471,31 @@ func (d *OpenStackDriver) GetVolNames(specs []corev1.PersistentVolumeSpec) ([]st
 	}
 	return names, nil
 }
+
+func waitForStatus(c *gophercloud.ServiceClient, id, pending []string, target []string, secs int) error {
+	return gophercloud.WaitFor(secs, func() (bool, error) {
+		current, err := servers.Get(c, id).Extract()
+		if err != nil {
+			return false, err
+		}
+
+		if strSliceContains(target, current.Status) {
+			return true, nil
+		}
+
+		if strSliceContains(pending, current.Status) {
+			return false, nil
+		}
+
+		return false, fmt.Errorf("unexpected status '%s', wanted target '%s'", current.Status, strings.Join(target, ", "))
+	})
+}
+
+func strSliceContains(haystack []string, needle string) bool {
```

Comment:
```suggestion
func containsStatus(allStatuses []string, wantedStatus string) bool {
```

Reply:
It's more common function, not only for status. I'd even move it to something like utils package. Can you suggest something?

Reply:
Ok, it's fine for now.